spec/models/reservation_spec.rb

```diff
@@ -413,6 +413,24 @@
             context "there is no following reservation" do
               it_behaves_like "a customer is allowed to edit"
             end
+
+            context "the instrument has a reservation lock window" do
+              let(:current_time) { Time.zone.now }
+
+              before :each do
+                instrument.update_attribute(:lock_window, 12)
+                reservation.reload
+              end
+
+              context "within the grace period" do
+                before do
+                  reservation.assign_attributes(reserve_start_at: current_time + 5.minutes, actual_start_at: current_time)
+                  reservation.save(validate: false)
```

Comment:
This pattern exists elsewhere, but I'm wondering if we could get away with `reservation.update_columns` on line 427 and make the non-validating `#save` unnecessary.